lib/events/impl.ex

```diff
@@ -0,0 +1,23 @@
+defmodule Events.Impl do
+  @moduledoc false
+
+  @callback child_spec(term()) :: Supervisor.child_spec()
+
+  @callback attach(
+              Events.handler_id(),
+              Events.event_prefix(),
+              module,
+              function :: atom,
+              config :: map
+            ) :: :ok | {:error, :already_exists}
+
+  @callback detach(Events.handler_id()) :: :ok | {:error, :doesnt_exist}
+
+  @callback list_attached_to(Events.event_name()) ::
```

Comment:
Maybe `list_handlers_by_name` and `list_handlers_by_prefix` below?

Reply:
Yeah, it makes sense 👍